apps/timetable/ui/js/index.js

```diff
@@ -196,6 +196,28 @@ define(['gh.core', 'gh.constants', 'gh.subheader', 'gh.calendar', 'gh.student-li
         });
     };
 
+    /**
+     * Open the external link
+     *
+     * @return {Boolean}    Return false to discard the natural link behaviour
+     */
+    var openExternalLink = function() {
```

Comment:
Could we reverse the flow a bit? We can call this function and sanitise the URL before rendering the template so when the click happens it's all default behaviour.